src/main/kotlin/com/openlattice/shuttle/Shuttle.kt

```diff
@@ -431,13 +497,28 @@ class Shuttle(
 
     fun launch(uploadBatchSize: Int): Long {
         val sw = Stopwatch.createStarted()
-        val total = flightPlan.entries.map { entry ->
-            logger.info("Launching flight: {}", entry.key.name)
-            val count = takeoff(entry.key, entry.value.getPayload(), uploadBatchSize, tableColsToPrint)
-            logger.info("Finished flight: {}", entry.key.name)
-            count
-        }.sum()
-        logger.info("Executed {} entity writes in flight plan in {} ms.", total, sw.elapsed(TimeUnit.MILLISECONDS))
+        var total = 0L
+        try {
+            total = flightPlan.entries.map { entry ->
+                val launchUpdate = "Launching flight: ${entry.key.name}"
+                writeLog(entry.key.name, launchUpdate, OffsetDateTime.now(), IntegrationStatus.IN_PROGRESS)
+
+                val tableColsToPrintForFlight = tableColsToPrint[entry.key] ?: listOf()
+                val count = takeoff(entry.key, entry.value.getPayload(), uploadBatchSize, tableColsToPrintForFlight)
+
+                val finishUpdate = "Finished flight: ${entry.key.name}"
+                writeLog(entry.key.name, finishUpdate, OffsetDateTime.now(), IntegrationStatus.SUCCEEDED)
+                count
+            }.sum()
```

Comment:
Can actually remove this `.sum()` as a synchronization point and use a `LongAdder` for total and just submit `count` to the `LongAdder` basically whenever. 
Food for thought, very likely unnecessary here but a cool pattern when the bottleneck is synch points :p